azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -34,13 +35,14 @@ func resourceArmKubernetesCluster() *schema.Resource {
 				profile := rawProfiles[0].(map[string]interface{})
 				networkPlugin := profile["network_plugin"].(string)
 
-				if networkPlugin == "kubenet" {
+				if networkPlugin == "kubenet" || networkPlugin == "azure" {
 					dockerBridgeCidr := profile["docker_bridge_cidr"].(string)
 					dnsServiceIP := profile["dns_service_ip"].(string)
 					serviceCidr := profile["service_cidr"].(string)
 
-					if dockerBridgeCidr == "" || dnsServiceIP == "" || serviceCidr == "" {
-						return fmt.Errorf("If the `network_plugin` is set to `kubenet` then the fields `docker_bridge_cidr`, `dns_service_ip` and `service_cidr` must not be empty.")
+					if !((dockerBridgeCidr == "" && dnsServiceIP == "" && serviceCidr == "") ||
+						(dockerBridgeCidr != "" && dnsServiceIP != "" && serviceCidr != "")) {
```

Comment:
_minor / not a blocker_ we could probably make this more readable by doing:
```
hasDockerCidr := dockerBridgeCidr != ""
hasServiceIP := dnsServiceIP != ""
hasServiceCidr := serviceCidr != ""
allEmpty := !hasDockerCidr && !hasServiceIP && !hasServiceCidr
allSet := hasDockerCidr && hasServiceIP && hasServiceCidr
if !allEmpty && !allSet {
  return fmt.Errorf("...")
}
```

Reply:
Good suggestion, let me figure out a better way.

---
In reply to: [207947438](https://github.com/terraform-providers/terraform-provider-azurerm/pull/1715#discussion_r207947438) [](ancestors = 207947438)